iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift

```diff
@@ -25,16 +25,33 @@ extension ConversationListCoordinator {
         }
     }
 
-    func presentThread(for channelId: ChannelId,
+    func presentThread(for cid: ConversationId,
                        messageId: MessageId,
                        startingReplyId: MessageId?) {
-        
-        let coordinator = ThreadCoordinator(with: channelId,
+
+        let coordinator = ThreadCoordinator(with: cid,
                                             messageId: messageId,
                                             startingReplyId: startingReplyId,
                                             router: self.router,
                                             deepLink: self.deepLink)
-        self.present(coordinator)
+        self.router.present(coordinator, source: self.conversationListVC)
+        self.addChildAndStart(coordinator) { _ in }
+    }
+
+    func presentMessageDetail(for channelId: ChannelId, messageId: MessageId) {
+        let message = Message.message(with: channelId, messageId: messageId)
+        let coordinator = MessageDetailCoordinator(with: message,
+                                                   router: self.router,
+                                                   deepLink: self.deepLink)
+        self.router.present(coordinator, source: self.conversationListVC)
+
+        self.addChildAndStart(coordinator) { [unowned self] message in
+            self.router.dismiss(source: self.conversationListVC) {
+                self.presentThread(for: channelId,
+                                      messageId: messageId,
+                                      startingReplyId: nil)
+            }
+        }
```

Comment:
New message detail coordinator. It returns the message if the user taps open thread. Then we dismiss and present the thread experience.

Reply:
Use the built in present call I made. It handles first responder better.

Reply:
Its specific to the list.